libzfs.pyx

```diff
@@ -3605,9 +3606,20 @@ cdef class ZFSResource(ZFSObject):
                 cur_prop.inherit(recursive=prop_details.get('recursive', False))
             else:
                 if 'value' in prop_details:
-                    props[prop_name] = prop_details['value']
+                    if self.properties.get(prop_name).value != prop_details.get('value') or \
```

Comment:
This caused a regression. `self.properties` can be `NoneType` which means this crashes with `AttributeError`